libpod/container_api.go

```diff
@@ -376,14 +376,18 @@ func (c *Container) Exec(tty, privileged bool, env, cmd []string, user string) e
 	logrus.Debugf("Successfully started exec session %s in container %s", sessionID, c.ID())
 
 	// Unlock so other processes can use the container
-	c.lock.Unlock()
-	locked = false
+	if !c.locked {
+		c.lock.Unlock()
+		locked = false
```

Comment:
I'm leery about this locked variable and potential (although admittedly tiny) of a timing issue.  It looks like the only place the locked variable is used is at line 278 to see if the container is locked in your defer statement.  Can you just use "if c.locked" there?  That seems safer to me.

Reply:
`c.locked` is a separate beast - it tracks whether the function needs to concern itself with locking at all. `locked` just indicates whether we have unlocked the container to allow things to happen while the exec session completes. The naming similarity is not good, though. I should probably rename `c.locked` to `c.batched` or similar

Reply:
Yes lets do that.

Reply:
Ah, thanks.  Yeah if you could swap out that variable name at some point, that would be nice.  Thx for the 411.